bitbots_vision_tools/scripts/colorpicker.py

```diff
@@ -0,0 +1,255 @@
+#!/usr/bin/env python3
+
+import cv2
+import numpy as np
+import time
+import pickle
+import os
+import rospy
+import VisionExtensions
+from cv_bridge import CvBridge
+from sensor_msgs.msg import Image
+
+
+class Colorpicker(object):
+    """
+    The bitbots_colorpicker node. Which is used to select colors that occur in the field.
+    """
+    def __init__(self):
+        rospy.init_node('bitbots_colorpicker')
+        rospy.loginfo('Initializing colorpicker...', logger_name="colorpicker")
+        self._cv_bridge = CvBridge()
+        self._image = np.zeros((300,300,3), dtype=np.uint8)
+
+        # Subscribe to Image-message
+        self._sub_image_msg = rospy.Subscriber(
+            "image_raw",
+            Image,
+            self._image_callback,
+            queue_size=1,
+            buff_size=60000000,
+            tcp_nodelay=False)
+
+        # Init
+        self._init_text()
+        cv2.namedWindow("Colorpicker")
+        cv2.setMouseCallback("Colorpicker", self._mouse_callback)
+        self._mouse_coord = (0,0)
+        self._left_click, self._undo_click = False, False
+        color_space = np.zeros((256, 256, 256), dtype=np.uint8)
+
+        # Default box size
+        self._box_size = 50
+
+        # Init history
+        self._history = [color_space]
+
+
+    def _mouse_callback(self, event, x, y, flags, param):
+        """
+        Callback for the OpenCV cursor.
+
+        :param event: Event type
+        :param x: Mouse x position
+        :param y: Mouse y position
+        :param flags: Callback flags
+        :param param: Some unused parameter
+        """
+        if flags == (cv2.EVENT_FLAG_SHIFTKEY + cv2.EVENT_FLAG_LBUTTON):
+            # Subtracts colors
+            self._undo_click = True
```

Comment:
> Please reset the click bools before or do something like:
>
> ``` self._undo_click = (flags == cv.EVENT_FLAG_SHIFTKEY + cv2.EVENT_FLAG_LBUTTON)```
> ```self._left_click = (not self._undo_click and event == cv2.EVENT_LBUTTONUP)```

Good point

Reply:
But the reset in the main loop is still required, because of the different update rates. Otherwise, the main loop adds the color space multiple times until the next mouse callback arrives.